generator/src/main/java/de/starwit/generator/services/AppCheckout.java

```diff
@@ -108,24 +112,21 @@ public AppTemplate checkoutAndUpdateAppTemplate(final DownloadAppTemplateDto dto
 		}
 
 		if (appTemplate.isCredentialsRequired()) {
-			//TODO: Sonderzeichencodierung
-			dto.setPassword(dto.getPass().replaceAll("@", "%40"));
-			srcDir = srcDir.replaceAll("://", "://" + dto.getUser() + ":" + dto.getPass() + "@");
-			LOG.info("Source directory is: " + srcDir);
+			dto.setPass(HtmlUtils.htmlEscape(dto.getPassword()));
+			srcDir = srcDir.replaceAll("://", "://" + dto.getUsername() + ":" + dto.getPass() + "@");

```

Comment:
What about the username? Could it not have special chars aswell?

Reply:
Will be done in AB#70

Reply:
also added special characters to username.